lucille-core/src/main/java/com/kmwllc/lucille/stage/ExtractEntitiesFST.java

```diff
@@ -0,0 +1,484 @@
+    this.stopOnHit = ConfigUtils.getOrDefault(config, "stop_on_hit", false);
+    this.ignoreCase = ConfigUtils.getOrDefault(config, "ignore_case", false);
+    this.entityField = ConfigUtils.getOrDefault(config, "entity_field", null);
+    this.updateMode = UpdateMode.fromConfig(config);
+    this.fileFetcher = new FileContentFetcher(config);
+
+    if (ignoreCase) {
+      this.analyzer = new StandardAnalyzer();
+    } else {
+      this.analyzer = new Analyzer() {
+        @Override
+        protected TokenStreamComponents createComponents(String fieldName) { return new TokenStreamComponents(new StandardTokenizer()); }
+      };
+    }
+  }
+
+  @Override
+  public void start() throws StageException {
+    StageUtils.validateFieldNumNotZero(sourceFields, "ExtractEntitiesFST");
+    StageUtils.validateFieldNumNotZero(destFields, "ExtractEntitiesFST");
+    StageUtils.validateFieldNumsSeveralToOne(sourceFields, destFields, "ExtractEntitiesFST");
+
+    try {
+      fileFetcher.startup();
+      if (usePayloads) {
+        buildFSTWithPayloads();
+      } else {
+        buildFSTNoPayloads();
+      }
+    } catch (IOException e) {
+      throw new StageException("Failed to initialize dictionaries for ExtractEntitiesFST", e);
+    }
+  }
+
+  @Override
+  public void stop() throws StageException {
+    fileFetcher.shutdown();
+  }
+
+  // Load and normalize terms with payloads
+  private LinkedHashMap<String, String> loadTermsAndPayloads() throws IOException, StageException {
+    LinkedHashMap<String, String> termToPayload = new LinkedHashMap<>();
+
+    // Read each dict in our list
+    for (String dictFile : dictionaries) {
+      try (CSVReader reader = new CSVReader(fileFetcher.getReader(dictFile))) {
+        String[] line;
+        while (true) {
+          try {
+            line = reader.readNext();
+          } catch (Exception e) {
+            throw new StageException("Error reading dictionary file: " + dictFile, e);
+          }
+
+          // End of file
+          if (line == null) {
+            break;
+          }
+
+          // Skip empty rows
+          if (line.length == 0 || line[0] == null) {
+            continue;
+          }
+
+          // Normalize term with trim and lowercase
+          String term = normalizeKey(line[0].trim());
+          if (term.isEmpty() || termToPayload.containsKey(term)) {
+            continue;
+          }
+
+          // Get the payload string
+          String payload = (line.length > 1 && line[1] != null && !line[1].trim().isEmpty())
+              ? line[1].trim()
+              : null;
+
+          termToPayload.put(term, payload);
+        }
+      }
+    }
+
+    return termToPayload;
+  }
+
+  private void buildFSTNoPayloads() throws IOException, StageException {
+    // Sort terms lexicographically
+    List<String> terms = new ArrayList<>(loadTermsAndPayloads().keySet());
+    terms.sort(Comparator.naturalOrder());
```

Comment:
sorting could be expensive for a huge dictionary -- might want to have a flag like dictionariesArePresorted -- defaults to false but let the user set it to true if they want to save startup time for dictionaries that are already sorted